src/app/shared/components/template/components/map/map.component.ts

```diff
@@ -0,0 +1,203 @@
+import { Component, OnInit } from "@angular/core";
+import { TemplateBaseComponent } from "../base";
+import { TemplateAssetService } from "../../services/template-asset.service";
+import { getParamFromTemplateRow, getStringParamFromTemplateRow } from "src/app/shared/utils";
+import { AppDataService } from "src/app/shared/services/data/app-data.service";
+import Map from "ol/Map";
+import View from "ol/View";
+import { OSM } from "ol/source";
+import TileLayer from "ol/layer/Tile";
+import VectorSource from "ol/source/Vector";
+import VectorLayer from "ol/layer/Vector";
+import GeoJSON from "ol/format/GeoJSON";
+import Style from "ol/style/Style";
+import Fill from "ol/style/Fill";
+import Stroke from "ol/style/Stroke";
+import chroma from "chroma-js";
+import BaseLayer from "ol/layer/Base";
+
+interface IMapLayer {
+  id: string;
+  description: string;
+  fill: string;
+  name: string;
+  opacity: number;
+  // a property of the dataset to be plotted
+  property: string;
```

Comment:
nit(non-blocking): better to use `/** */` comment blocks as these will be picked up by vscode intellisense when referring to the interface elsewhere